cpp/src/arrow/array/array_binary_test.cc

```diff
@@ -106,6 +106,13 @@ class TestStringArray : public ::testing::Test {
     AssertZeroPadded(*strings_);
   }
 
+  void TestArrayIndexOperator() {
+    auto& arr = *strings_;
```

Comment:
Should use `const auto&`, but really there doesn't seem to be a need for this, is there?